pandas/core/sorting.py

```diff
@@ -319,6 +319,33 @@ def nargsort(
     return indexer
 
 
+def nargminmax(values, method: str):
+    """
+    Implementation of np.argmin/argmax but for ExtensionArray and which
+    handles missing values.
+
+    Parameters
+    ----------
+    values : ExtensionArray
+    method : {"argmax", "argmin"}
+
+    Returns
+    -------
+    int
+    """
+    assert method in {"argmax", "argmin"}
+    func = np.argmax if method == "argmax" else np.argmin
+
+    mask = np.asarray(isna(values))
+    values = values._values_for_argsort()
```

Comment:
im not wild about relying on non-public attrs here.  could we have the EA method pass `values` and `mask`?

Reply:
`_values_for_argsort` *is* a "public developer" API (it's part of the EA interface), that's the entire point of it. 

I know we have had the discussion about the point of `_values_for_argsort` and in principle we could also do without. But at this point, we have that method, it is used for `argsort` as well, so I think it is most logical that I use it here. And we can continue that general discussion about `_values_for_argsort` elsewhere.